client/changes.go

```diff
@@ -157,3 +157,31 @@ func (client *Client) Changes(opts *ChangesOptions) ([]*Change, error) {
 
 	return chgs, err
 }
+
+// WaitChangeOptions holds the options for the WaitChange call.
+type WaitChangeOptions struct {
+	// If non-zero, wait at most this long before returning the current change
+	// data. The timeout elapsing is not considered an error, so if a timeout
+	// is specified, the caller should check Change.Ready to determine whether
+	// the change is actually finished.
```

Comment:
Yeah, Harry and I went back and forth on that and ended up that it shouldn't be an error, primarily because when Pebble returns an error it can't also return other data (the Change in this case), and I thought it might be useful to have the current state of the change at the time of the timeout.

However, on reflection it makes for a less error-prone API if it's just an error, and in the success case the change is always ready. If the caller really wants the current state of a change in the case of a timeout/error, they can always make an additional call to `client.Change()` to get the current status. However, it's unlikely you'll want to.

In any case, I've used HTTP status code `504 Gateway Timeout` for this (along with a reasonable error message). It's not quite a perfect semantic match, but it seems close enough.